src/UI/Buyer/src/app/app.module.ts

```diff
@@ -540,6 +544,30 @@ export class AppModule {
     this.buildWebComponent(OCMLocationManagement, 'ocm-location-management')
   }
 
+  configureTranslationService(): void {
+    const browserCultureLang = this.translate.getBrowserCultureLang();
+    const browserLang = this.translate.getBrowserLang();
+    
+    this.translate.addLangs(this.appConfig.supportedLanguages)
+    const languages = this.translate.getLangs()
+    const selectedLang = this.cookieService.getObject(this.languageCookieName)
+    if (selectedLang && languages.includes(selectedLang.toString())) {
+      this.translate.setDefaultLang(selectedLang.toString());
+    } else if (languages.includes(browserCultureLang)) {
+      this.translate.setDefaultLang(browserCultureLang);
+    } else if (languages.includes(browserLang)) {
+      this.translate.setDefaultLang(browserLang);
+    } else if (languages.includes(this.appConfig.defaultLanguage)) {
+      this.translate.setDefaultLang(this.appConfig.defaultLanguage)
+    } else if (languages.length > 0) {
+      this.translate.setDefaultLang(languages[0])
```

Comment:
Can we throw an error at the top of this function if there are no languages?

Reply:
Sure, I will add defensive programming here. I wasn't sure if we were going to fail fast or just silently handle this by letting the placeholder references display. Are you aware of a preferred approach or whether their are any post-deployments checks to identify the app is running successfully?

Reply:
Errors are now thrown when `supportedLanguages` or `defaultLanguage` are not defined in the config, or when no language can be identified for use (which should never occur).

The initial errors do prevent the sites from loading, so we may want to handle showstopping errors by rendering some kind of technical difficulties error page, however I am not sure the best way to manage this at this time.